model/lifecycle.go

```diff
@@ -845,9 +852,24 @@ func TryMarkPatchBuildFinished(b *build.Build, finishTime time.Time, updates *St
 	return nil
 }
 
+func getTaskCreateTime(project *Project, v *version.Version) (time.Time, error) {
```

Comment:
I would have this take in a project ID as a string since you don't use any other properties of it